executor/merge_join.go

```diff
@@ -46,7 +48,7 @@ type MergeJoinExec struct {
 	innerTable *mergeJoinInnerTable
 	outerTable *mergeJoinOuterTable
 
-	innerRows     []chunk.Row
+	innerRows     *chunk.RowContainer
 	innerIter4Row chunk.Iterator
```

Comment:
You should check `innerIter4Row.Error()` somewhere.

Reply:
It is checked inside joiner.go.